src/persistence.cr

```diff
@@ -96,16 +97,17 @@ module Rome
         self.attributes = attributes
       end
 
-      if changed?
-        if self.responds_to?(:updated_at=)
-          self.updated_at = Time.utc
-        end
-
-        self.class.update(id, attributes_for_update.not_nil!)
+      save_associations do
```

Comment:
Should probably be inside a transaction? In which case the `chanes_applied` call would be well placed (outside the `save_associations` block). Otherwise, without a transaction, the `changes_applied` should be called immediately after the update.